src/push-state-tree.js

```diff
@@ -164,194 +228,210 @@ function PushStateTree(options) {
+      var uri = internalHistory.last().url;
+      return !this.basePath || uri.indexOf(this.basePath) === 0;
     }
   });
 
-  this.eventStack = {
-    leave: [],
-    change: [],
-    enter: [],
-    match: []
-  };
-
-  root.addEventListener(POP_STATE, () => {
-    var eventURI = this.uri;
-    var eventState = this.state;
-    this.rulesDispatcher();
-
-    oldURI = eventURI;
-    oldState = eventState;
-
-    // If there is holding dispatch in the event, do it now
-    if (holdingDispatch) {
-      this.dispatch();
+  let disabled = true;
+  let disableMethod;
+  Object.defineProperty(this, 'disabled', {
+    get() {
+      return disabled;
+    },
+    set(value) {
+      value = value === true;
+      if (value != disabled) {
+        disabled = value;
+        if (disabled) disableMethod();
+        else disableMethod = this.startGlobalListeners();
+      }
     }
   });
+  this.disabled = options.disabled === true;
 
-  var readOnhashchange = false;
-  var onhashchange = () => {
-    // Workaround IE8
-    if (readOnhashchange) return;
-
-    // Don't dispatch, because already have dispatched in popstate event
-    if (oldURI === this.uri) return;
-
-    var eventURI = this.uri;
-    var eventState = this.state;
-    this.rulesDispatcher();
-
-    oldURI = eventURI;
-    oldState = eventState;
-
-    // If there is holding dispatch in the event, do it now
-    if (holdingDispatch) {
-      this.dispatch();
+  // Setup options, must be executed after define all properties
+  for (let prop in options) {
+    if (options.hasOwnProperty(prop)) {
+      this[prop] = options[prop];
     }
-  };
-
-  this.avoidHashchangeHandler = () => {
-    // Avoid triggering hashchange event
-    root.removeEventListener(HASH_CHANGE, onhashchange);
-    readOnhashchange = true;
-  };
-
-  root.addEventListener(HASH_CHANGE, onhashchange);
-
-  // Uglify propourses
-  var dispatchHashChange = () => {
-    root.dispatchEvent(new HashChangeEvent(HASH_CHANGE));
-  };
-
-  // Modern browsers
-  document.addEventListener('DOMContentLoaded', dispatchHashChange);
-  // Some IE browsers
-  root.addEventListener('readystatechange', dispatchHashChange);
-  // Almost all browsers
-  root.addEventListener('load', () => {
-    dispatchHashChange();
-
-    if (!isIE()) return;
-
-    setInterval(() => {
-      if (this.uri !== oldURI) {
-        dispatchHashChange();
-        return;
-      }
-      if (readOnhashchange) {
-        readOnhashchange = false;
-        oldURI = this.uri;
-        root.addEventListener(HASH_CHANGE, onhashchange);
-      }
-    }, 50);
-  });
+  }
 
   return this;
 }
 
-/*eslint no-unused-vars: ["error", { "varsIgnorePattern": "oldState" }]*/
-var oldState = null;
-var oldURI = null;
 var eventsQueue = [];
 var holdingDispatch = false;
 var holdDispatch = false;
 
 let hasPushState = !!(history && history.pushState);
 
-Object.assign(PushStateTree, {
+let mixinPushStateTree = {
   VERSION,
```

Comment:
Webpack will change this for a string with the version while building it. It's in the define plugin, I also setup the eslint to provide as global and skip the warning.

I will add a comment :)